### PR TITLE
AO - Assignment 08

### PR DESCRIPTION
**Pull request by Gustl22**
Tried to do assignment 8 and was immediately set back as the skeleton wasn't working out of the box (which unnecessarily takes time to solve for every student):

- [webshot ](https://www.npmjs.com/package/node-webshot) not [compatible ](https://github.com/brenden/node-webshot/issues/240) with latest node version (12 and 13).
- DBHandler not creating user table (although already implemented)
- old packages

So for the first issue, without needing to downgrade or use special helper to switch node versions, it is the best to use another screenshot library, as webshot won't be updated anymore. So I propose [capture-website](https://github.com/sindresorhus/capture-website) for that.

Second: I wasn't quite successful to get the meaning of the connection states of mysql for js (more precisely [authenticated ](https://github.com/mysqljs/mysql/blob/32a0293e6d7dbb6132d89ddf261dcd17bb50c080/lib/Connection.js#L436)). But whenever I connected to the db, the connection state already was in `connected` state. So users table actually never was created.

Third: I upgraded some packages to avoid incompatibilites.

All issues are fixed in 563ebbd26a9197d0855461878e9269c0b36b4a2a . I think if using this assignment the next year you may want consider updating the skeleton to not frustrate students minds :stuck_out_tongue_winking_eye:

I also propose to add a `.gitignore` to root in order to exclude all IDE's stuff that can be very annoying (excluding these files before each commit or branch checkout).
```
# IDEs
.idea
.vscode
```

### CODE REVIEW

**Review by changkun — APPROVED** [review]
LGTM

- The updates of the skeleton need to be confirmed by @Flo890 
- The `.gitignore` can exclude all IDE's. However, students can always add `.gitignore` in their solution folder to avoid this (do not open the entire repo as a working directory, instead, only working on their solution directory).

**Comment by Gustl22** [comment]
Thanks, actually wasn't aware, that IDE recognizes the git root (outside its scope), when opening subdirectory, in order to take advantage of the VCS.

File: assignments/solutions/AO-Assignment08/ScreenshotApp/package.json [hunk 1]
```diff
@@ -3,21 +3,22 @@
   "version": "0.0.0",
   "private": true,
   "scripts": {
-    "start": "node ./bin/www"
+    "start": "nodemon ./bin/www"
```
Comment (Gustl22) [inline assignments/solutions/AO-Assignment08/ScreenshotApp/package.json:6]: forgot to revert to node, though it's quite comfortable to code with nodemon 😉

**Review by changkun — APPROVED** [review]
I am merging this since the changes regarding the skeleton do not exist in this PR.
@Flo890 might need to check the discussion here for further decisions.

**Pull request merged**